pathology/hovernet/infer_torch.py

```diff
@@ -0,0 +1,256 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from net import HoVerNet
+from transforms import (
+    GenerateWatershedMaskd,
+    GenerateInstanceBorderd,
+    GenerateDistanceMapd,
+    GenerateWatershedMarkersd,
+    Watershedd,
+    GenerateInstanceContour,
+    GenerateInstanceCentroid,
+    GenerateInstanceType,
+)
+
+import os
+import glob
+import torch
+import numpy as np
+from argparse import ArgumentParser
+from monai.data import DataLoader, decollate_batch, Dataset, CacheDataset
+from monai.metrics import DiceMetric
+from monai.metrics.confusion_matrix import ConfusionMatrixMetric
+from monai.losses import DiceLoss
+# from monai.networks.nets import HoVerNet
+from monai.transforms import (
+    Activations,
+    AsDiscrete,
+    AsDiscreted,
+    Compose,
+    ScaleIntensityRanged,
+    CastToTyped,
+    Lambdad,
+    SplitDimd,
+    EnsureChannelFirstd,
+    ComputeHoVerMapsd,
+    RandFlipd,
+    RandRotate90d,
+    RandGaussianSmoothd,
+    GaussianSmooth,
+    FillHoles,
+    BoundingRect,
+    CenterSpatialCropd,
+    SaveImage,
+)
+# from monai.apps.pathology.transforms.post import (
+#     GenerateInstanceContour,
+#     GenerateInstanceCentroid,
+#     GenerateInstanceType
+# )
+
+from monai.utils import set_determinism, convert_to_tensor
+from monai.utils.enums import HoVerNetBranch
+from monai.visualize import plot_2d_or_3d_image
+from skimage import measure
+
+
+def prepare_data(data_dir, phase):
+    data_dir = os.path.join(data_dir, phase)
+
+    files = sorted(
+        glob.glob(os.path.join(data_dir, "*/*.npy")))
+
+    images, labels, inst_maps, type_maps = [], [], [], []
+    for file in files:
+        data = np.load(file)
+        images.append(data[..., :3].transpose(2, 0, 1))
+        inst_maps.append(measure.label(data[..., 3][None]).astype(int))
+        type_maps.append(data[..., 4][None])
+        labels.append(np.array(data[..., 3][None] > 0, dtype=int))
+
+    data_dicts = [
+        {"image": _image, "label": _label, "label_inst": _inst_map, "label_type": _type_map}
+        for _image, _label, _inst_map, _type_map in zip(images, labels, inst_maps, type_maps)
+    ]
+
+    return data_dicts
+
+
+def _dice_info(true, pred, label):
+    true = np.array(true == label, np.int32)
+    pred = np.array(pred == label, np.int32)
+    inter = (pred * true).sum()
+    total = (pred + true).sum()
+    return inter, total
+
+
+def post_process_WS(output, device, return_binary=True, return_centroids=False, output_classes=None):
```

Comment:
This can be a post transform

Reply:
Hi @SachidanandAlle, thanks for your advice, I have added it in this PR https://github.com/Project-MONAI/MONAI/pull/5462.